File: raulG91.js
//Arimethic operators

let a = 2;
let b = 3;

//Sum 
let sum = a+b;
console.log("Sum: ", sum);

//Diff
let diff = a-b;
console.log("Diff: ", diff);

//Multiply 
let multiply = a*b;
console.log("Mltiply: ",multiply);

//Division 
let division = b/a;
console.log("Division: ",division);

//Remainder
let remainder = b%a;
console.log("Module: ", remainder);

//Pow 
let pow = a**b;
console.log("Pow: ",pow);

//Logical operators
console.log("***Logical operators***");

//And
let and = true && false;
console.log("And: ", and);

//OR
let or = false || true;
console.log("Or: ", or);

//Not
let not = !(true);
console.log("Not: ", not);

//Comparassion
console.log("***Comparassion***");

//Equal 
let equal = a==b;
console.log("Equal: ",equal);

//Not equal
let not_equal = a!=b;
console.log("Not equal: ", not_equal);

//Bigger
let bigger = a > b;
console.log("Bigger: ",bigger);

//Bigger or equal
let bigger_equal = a >= b;
console.log("Bigger or equal: ", bigger_equal);

//Lower 
let lower = a < b;
console.log("Lower: ", lower);

//Lower or equal
let lower_equal = a <= b;
console.log("Lower or equal: ", lower_equal);

//Assignation
let c = 10;
console.log("= ", c);

//Beloning
let array = [3,5,0,1];
let belong = 3 in array;
console.log("IN: ",belong);

//Identity
let identity = a ===b;
console.log("Identity: ",identity);

//Bits 
console.log("***Bits***");

//And
let bits_and = 5 & 3;
console.log("&: ",bits_and);

//Or
let bits_or = 5 | 3;
console.log("|: ", bits_or);

// XOR
let bits_xor = 5 ^ 3;
console.log("^: ",bits_xor);

// Not
let bits_not = ~ 5;
console.log("~: ",bits_not);

// Control 
console.log("***If statements***")
//If
if(a<b){
    console.log("If statement")
}
else{
    console.log("Else statement");
}
//inline if
(a<b)? console.log("If with ?"):console.log("else with ?");

//Switch

let color = "orange";

switch(color){
    case "orange":
        console.log("orange color");
        break;
    case "blue":
        console.log("blue color");
        break;
    default: 
        console.log("Default option") ;
        break;       
}


//Loops
console.log("***While***");

while(true){

    console.log("While loop");
    break;
}

console.log("***for loop***");

for(let i = 0;i<3;i++){

    console.log(i);
}

//Do while loop
console.log("***DO while***");

let j=0;

do{
    console.log(j);
    j++;

}while(j<2);

//Exception  
console.log("***Exception***")
try{

  f
}catch(err){
    console.log(err.message);
}

//Extra exercise

for(let index = 10; index <=55;index++){

    if(index%2==0 && index !=16 && index%3 != 0){
  
      console.log(index);
    }
  }

File: Roadmap/01 - OPERADORES Y ESTRUCTURAS DE CONTROL/python/raulG91.py

#Arithmetic operations
sum = 2+3
print("Sum: ", sum)
diff =  5-1
print("Diff: ", diff)
multiply = 6*8
print("multiply: ", multiply)
division =  40 / 5
print("Division", division)
int_division = 3//2
print("Int division", int_division)
module = 30 % 2
print("Module", module)
power = 2**2
print("Power: ",str(power))

#Logical 
v_and = True and False
print("And: ",v_and)
v_or = True or False
print("Or : ", v_or)
v_not = not(True)
print("Not: ", v_not)

#Comparation
equal = True == True
print("Equal : ", equal)
not_equal = False != True
print("Not equal: ", not_equal)
greater = 3 > 2
print("> : ", greater)
greater_equal = 3>=3
print(">= :", greater_equal)
less = 5 < 10 
print("<: ", less)
less_equal = 5<= 2
print("<= : ",less_equal)

#Identity
identity = 1 is -1
print("is : ", identity)

#Belonging
beloning = 1 in [3,4,5,1]
print("in: ",beloning)

#Bits
bits_and = 1 & 1
print("& : ", bits_and)
bits_or = 1 | 1
print("| : ", bits_or)
bits_xor = 1 ^ 1
print("^ : ", bits_xor)
bits_not = ~ 1
print("~ : ",bits_not)

a = 5

#if - elif - else
if a < 3:
    print("Number less than 3")
elif a>=3 and a <10:
    print("Number between 3 and 9")
else:
    print("Number bigger than 10")  
    
#Loops    
for number in range(0,5):
    print(number)          

while (True):
    print("While loop")
    break    

try:
    n = float(input("Enter a number: "))
    m = 4
    print("{}/{} = {}".format(n,m,n/m))
except:
    print("Number entered is not correct")
else:
    print("Everything is ok")
       
'''
Extra Exercise: 
* Crea un programa que imprima por consola todos los números comprendidos
* entre 10 y 55 (incluidos), pares, y que no son ni el 16 ni múltiplos de 3.
'''        

for num in range(10,56):
    if (num % 2 == 0) and (num !=16)  and (num % 3 != 0):
        print(num)
    